### ISSUE TITLE
building for ios without build number

### ISSUE THREAD

**Issue by RobinManoli** [flutter/flutter#25401]
tldr: Building in ios should require a build number, or at least issue a warning.

In the `pubspec.yaml` info about the version of the Flutter project it is stated that the version number is "followed by an optional build number separated by a +."

code in pubspec.yaml
```
# So the +1 here is optional according to flutter
# but appstore will not accept your build without a build number
version: 1.0.0+1
```

The thing is that you cannot upload a build to apple if it doesn't have a build number. So after you successfully spend several minutes to build your project , and then spend several minutes to archive it in xcode, then verify it (according to flutter documentation) and then upload it.... then apple tells you a cryptic error meaning you didn't set your build number so you have to repeat this process again, every time you didn't set a build number.

**Comment by spidgorny** [comment]
@zoechi Could you share the script with us? It's a pain to use Codemagic and stumble on the same version.

**Comment by kangwang1988** [comment]
https://github.com/flutter/flutter/pull/27743/files  seems related.
Welcome to comment on the pr so we can do it better.
@spidgorny  @RobinManoli

**Comment by jmagman** [comment]
I have a PR that will use the entire version if there is no + present.
https://github.com/flutter/flutter/pull/37036

Though it's a good idea to output a warning if a build number could not be parsed.

**Comment by jmagman** [comment]
> Though it's a good idea to output a warning if a build number could not be parsed.

*a warning visible from `flutter run`, not Xcode.

**Comment by jmagman** [comment]
Please let us know if https://github.com/flutter/flutter/pull/37036 does not fix the issue.  Closing.
